\documentclass{article}\usepackage{graphicx, color}
%% maxwidth is the original width if it is less than linewidth
%% otherwise use linewidth (to make sure the graphics do not exceed the margin)
\makeatletter
\def\maxwidth{ %
  \ifdim\Gin@nat@width>\linewidth
    \linewidth
  \else
    \Gin@nat@width
  \fi
}
\makeatother

\definecolor{fgcolor}{rgb}{0.2, 0.2, 0.2}
\newcommand{\hlnumber}[1]{\textcolor[rgb]{0,0,0}{#1}}%
\newcommand{\hlfunctioncall}[1]{\textcolor[rgb]{0.501960784313725,0,0.329411764705882}{\textbf{#1}}}%
\newcommand{\hlstring}[1]{\textcolor[rgb]{0.6,0.6,1}{#1}}%
\newcommand{\hlkeyword}[1]{\textcolor[rgb]{0,0,0}{\textbf{#1}}}%
\newcommand{\hlargument}[1]{\textcolor[rgb]{0.690196078431373,0.250980392156863,0.0196078431372549}{#1}}%
\newcommand{\hlcomment}[1]{\textcolor[rgb]{0.180392156862745,0.6,0.341176470588235}{#1}}%
\newcommand{\hlroxygencomment}[1]{\textcolor[rgb]{0.43921568627451,0.47843137254902,0.701960784313725}{#1}}%
\newcommand{\hlformalargs}[1]{\textcolor[rgb]{0.690196078431373,0.250980392156863,0.0196078431372549}{#1}}%
\newcommand{\hleqformalargs}[1]{\textcolor[rgb]{0.690196078431373,0.250980392156863,0.0196078431372549}{#1}}%
\newcommand{\hlassignement}[1]{\textcolor[rgb]{0,0,0}{\textbf{#1}}}%
\newcommand{\hlpackage}[1]{\textcolor[rgb]{0.588235294117647,0.709803921568627,0.145098039215686}{#1}}%
\newcommand{\hlslot}[1]{\textit{#1}}%
\newcommand{\hlsymbol}[1]{\textcolor[rgb]{0,0,0}{#1}}%
\newcommand{\hlprompt}[1]{\textcolor[rgb]{0.2,0.2,0.2}{#1}}%

\usepackage{framed}
\makeatletter
\newenvironment{kframe}{%
 \def\at@end@of@kframe{}%
 \ifinner\ifhmode%
  \def\at@end@of@kframe{\end{minipage}}%
  \begin{minipage}{\columnwidth}%
 \fi\fi%
 \def\FrameCommand##1{\hskip\@totalleftmargin \hskip-\fboxsep
 \colorbox{shadecolor}{##1}\hskip-\fboxsep
     % There is no \\@totalrightmargin, so:
     \hskip-\linewidth \hskip-\@totalleftmargin \hskip\columnwidth}%
 \MakeFramed {\advance\hsize-\width
   \@totalleftmargin\z@ \linewidth\hsize
   \@setminipage}}%
 {\par\unskip\endMakeFramed%
 \at@end@of@kframe}
\makeatother

\definecolor{shadecolor}{rgb}{.97, .97, .97}
\definecolor{messagecolor}{rgb}{0, 0, 0}
\definecolor{warningcolor}{rgb}{1, 0, 1}
\definecolor{errorcolor}{rgb}{1, 0, 0}
\newenvironment{knitrout}{}{} % an empty environment to be redefined in TeX

\usepackage{alltt}
\usepackage[sc]{mathpazo}
\usepackage[T1]{fontenc}
\usepackage{geometry}
\geometry{verbose,tmargin=2.5cm,bmargin=2.5cm,lmargin=2.5cm,rmargin=2.5cm}
\setcounter{secnumdepth}{2}
\setcounter{tocdepth}{2}
\usepackage{url}
\usepackage[unicode=true,pdfusetitle,
 bookmarks=true,bookmarksnumbered=true,bookmarksopen=true,bookmarksopenlevel=2,
 breaklinks=false,pdfborder={0 0 1},backref=false,colorlinks=false]
 {hyperref}
\hypersetup{
 pdfstartview={XYZ null null 1}}
\usepackage{breakurl}
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\begin{document}




\author{Ricky Lim}
\title{PCA Check after Gem-Mapping}
\maketitle

\begin{verbatim}
  Filename: PCACheck.Rnw 
  Working directory: /TEMP_DDN/users/gfilion/rlim/H1_ColorChromatin/PCACheckGemMapH1 
\end{verbatim}

\section{Load the input table}

\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
bigTable <- \hlfunctioncall{read.delim}(\hlstring{"input/H1_matBin.bed"})
\hlfunctioncall{head}(bigTable)
\hlfunctioncall{dim}(bigTable)

matTable <- bigTable[, 4:162]
\hlfunctioncall{head}(matTable)
\hlfunctioncall{dim}(matTable)
\end{alltt}
\end{kframe}
\end{knitrout}


\section{Load the Annotation's file}

\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
annot <- \hlfunctioncall{read.delim}(\hlstring{"idH1_Fastq_annotation.txt"}, header = F)
\hlfunctioncall{dim}(annot)
\hlfunctioncall{head}(annot)[10:13]
\hlfunctioncall{tail}(annot)[10:13]

\hlcomment{# grep the file numbers}
id_ <- \hlfunctioncall{sub}(\hlstring{".*-(\textbackslash{}\textbackslash{}d\{3\}|\textbackslash{}\textbackslash{}d\{3\}[ab]).*"}, \hlstring{"\textbackslash{}\textbackslash{}1"}, annot$V11)
\hlfunctioncall{print}(id_)
\hlfunctioncall{length}(id_)

\hlcomment{# double check for the id mapping from annotation file with the colnames of the matrix}
annot_dataset <- \hlfunctioncall{paste}(\hlstring{"X"}, id_, sep = \hlstring{""}) %in% \hlfunctioncall{colnames}(matTable)

\hlcomment{# get the samples given the file no}
sample_dataset <- annot$V13[annot_dataset]
\hlfunctioncall{head}(sample_dataset)[1:5]
\hlfunctioncall{length}(sample_dataset)

\hlcomment{# function to get the file number in the loaded dataset given the sampleName}
getProfileId <- \hlfunctioncall{function}(sampleName) \{
    \hlfunctioncall{library}(stringr)
\hlcomment{    # get only the partial match (in case!)}
    sample_dataset <- \hlfunctioncall{str_extract}(sample_dataset, sampleName)
    sample_ <- (sample_dataset == sampleName)
    sample_no <- id_[annot_dataset][sample_]
    sample_no <- sample_no[!\hlfunctioncall{is.na}(sample_no)]
    sample_names <- \hlfunctioncall{paste}(\hlstring{"X"}, sample_no, sep = \hlstring{""})
    \hlfunctioncall{return}(sample_names)
\}

\hlcomment{# e.g}
\hlfunctioncall{getProfileId}(\hlstring{"Input"})
\hlfunctioncall{getProfileId}(\hlstring{"CtBP2"})

\end{alltt}
\end{kframe}
\end{knitrout}



\section{Assigning NAs}
NAs were assigned for rows(genomic coordinates) in which in all profiles they were no reads

\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
matTable[\hlfunctioncall{which}(\hlfunctioncall{rowSums}(matTable) == 0), ] <- NA
\hlfunctioncall{dim}(matTable)
\end{alltt}
\end{kframe}
\end{knitrout}


\section{matTable no NAs}

\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
matTableNoNA <- matTable[\hlfunctioncall{complete.cases}(matTable), ]
\hlfunctioncall{sum}(\hlfunctioncall{rowSums}(matTableNoNA[, ]) == 0)
\hlfunctioncall{head}(matTableNoNA)
\hlfunctioncall{nrow}(matTableNoNA)/\hlfunctioncall{nrow}(matTable)
\hlfunctioncall{dim}(matTableNoNA)
\end{alltt}
\end{kframe}
\end{knitrout}


\section{Create PCA Object}

\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
log_mat <- \hlfunctioncall{log}(matTableNoNA + 1)
pca_mat <- \hlfunctioncall{prcomp}(log_mat, scale. = T)
\hlfunctioncall{plot}(pca_mat, main = \hlstring{"Scree Plot"})
\end{alltt}
\end{kframe}
\end{knitrout}


\section{PCA Labs}

\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlcomment{# order the PCA rotation matrix}
lab_PCA <- pca_mat$rotation[\hlfunctioncall{order}(\hlfunctioncall{rownames}(pca_mat$rotation)), ]
lab_PCA <- \hlfunctioncall{as.data.frame}(lab_PCA)
\hlfunctioncall{head}(lab_PCA)
lab_PCA[140:159, 1:5]
\hlfunctioncall{row.names}(lab_PCA) <- \hlfunctioncall{rownames}(lab_PCA)


\hlcomment{# get the lab's info}
lab_info <- annot$V1[annot_dataset]
lab_info

\hlcomment{# plot lab's effect without inputs}
\hlfunctioncall{pdf}(\hlstring{"figs/lab_effect.pdf"}, useDingbats = FALSE)
\hlfunctioncall{plot}(lab_PCA[, 1], lab_PCA[, 2], col = \hlfunctioncall{c}(\hlstring{"orange"}, \hlstring{"green"}, \hlstring{"red"})[lab_info], pch = 19, xlab = \hlstring{"PC1"}, 
    ylab = \hlstring{"PC2"}, frame = F, main = \hlstring{"Lab Effect"})
\hlfunctioncall{legend}(y = 0.2, x = 0.04, pch = 19, cex = 1, col = \hlfunctioncall{c}(\hlstring{"orange"}, \hlstring{"green"}, \hlstring{"red"}), legend = \hlfunctioncall{levels}(lab_info), 
    box.lwd = 0, box.col = \hlstring{"white"}, bg = \hlstring{"white"})
\hlfunctioncall{dev.off}()

\hlcomment{# plot lab's effect with inputs}
\hlfunctioncall{pdf}(\hlstring{"figs/Inputslab_effect.pdf"}, useDingbats = FALSE)
\hlfunctioncall{plot}(lab_PCA[, 1], lab_PCA[, 2], col = \hlfunctioncall{c}(\hlstring{"orange"}, \hlstring{"green"}, \hlstring{"red"})[lab_info], pch = 19, xlab = \hlstring{"PC1"}, 
    ylab = \hlstring{"PC2"}, frame = F, main = \hlstring{"Lab Effect"})
All_Input <- \hlfunctioncall{c}(\hlfunctioncall{getProfileId}(\hlstring{"RevXlinkChromatin"}), \hlfunctioncall{getProfileId}(\hlstring{"Input"}))
\hlfunctioncall{points}(lab_PCA[All_Input, 1], lab_PCA[All_Input, 2], col = \hlstring{"blue"}, cex = 1.5)

\hlfunctioncall{points}(lab_PCA[All_Input, 1], lab_PCA[All_Input, 2], col = \hlstring{"blue"}, cex = 1.5)
\hlfunctioncall{legend}(y = 0.2, x = 0.04, pch = 19, cex = 1, col = \hlfunctioncall{c}(\hlstring{"orange"}, \hlstring{"green"}, \hlstring{"red"}), legend = \hlfunctioncall{levels}(lab_info), 
    box.lwd = 0, box.col = \hlstring{"white"}, bg = \hlstring{"white"})

\hlfunctioncall{legend}(y = 0.1, x = 0.04, pch = 1, cex = 1, col = \hlfunctioncall{c}(\hlstring{"blue"}), legend = \hlfunctioncall{c}(\hlstring{"input"}), box.lwd = 0, 
    box.col = \hlstring{"white"}, bg = \hlstring{"white"})
\hlfunctioncall{text}(lab_PCA[All_Input, 1], lab_PCA[All_Input, 2], All_Input, pos = 1)
\hlfunctioncall{dev.off}()


\end{alltt}
\end{kframe}
\end{knitrout}

\begin{figure*}
  \includegraphics{figs/lab_effect.pdf}
\end{figure*}
\pagebreak

\begin{figure*}
  \includegraphics{figs/Inputslab_effect.pdf}
\end{figure*}
\pagebreak

X111 and X109, were on top of one another, suggesting possible duplicated samples.

\subsection{Duplicated Samples}
\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlfunctioncall{cor}(matTableNoNA[, \hlstring{"X109"}], matTableNoNA[, \hlstring{"X111"}], method = \hlstring{"spearman"})
\end{alltt}
\begin{verbatim}
## [1] 1
\end{verbatim}
\end{kframe}
\end{knitrout}


X111 and X109, were duplicated. These samples that were uploaded two times in ENCODE were similar.

\section{Metainfo}
\begin{knitrout}
\definecolor{shadecolor}{rgb}{0.969, 0.969, 0.969}\color{fgcolor}\begin{kframe}
\begin{alltt}
\hlfunctioncall{sessionInfo}()
\end{alltt}
\begin{verbatim}
## R version 2.15.0 (2012-03-30)
## Platform: x86_64-unknown-linux-gnu (64-bit)
## 
## locale:
## [1] C
## 
## attached base packages:
## [1] stats     graphics  grDevices datasets  utils     methods   base     
## 
## other attached packages:
## [1] Cairo_1.5-1       codetools_0.2-8   knitr_1.2         stringr_0.6      
## [5] vimcom_0.9-8      setwidth_1.0-3    cacheSweave_0.6-1 stashR_0.3-5     
## [9] filehash_2.2-1   
## 
## loaded via a namespace (and not attached):
## [1] digest_0.5.2   evaluate_0.4.3 formatR_0.7    plyr_1.7.1     tools_2.15.0
\end{verbatim}
\end{kframe}
\end{knitrout}





\end{document}

